packages/build-utils/src/get-project-paths.ts

```diff
@@ -0,0 +1,75 @@
+import { detectFramework } from './detect-framework';
+import { DetectorFilesystem } from './detectors/filesystem';
+import frameworks from '@vercel/frameworks';
+
+const MAX_DEPTH_TRAVERSE = 3;
+
+export interface GetProjectPathsOptions {
+  fs: DetectorFilesystem;
+  path?: string;
+  skipPaths?: string[];
+  depth?: number;
+}
+
+export type ProjectPath = string;
+
+export const getProjectPaths = async ({
+  fs,
+  path,
+  skipPaths,
+  depth = 1,
+}: GetProjectPathsOptions): Promise<ProjectPath[]> => {
+  const allPaths: Array<ProjectPath> = [];
+  if (path && skipPaths?.includes(path)) {
+    return allPaths;
+  }
+
+  if (depth < MAX_DEPTH_TRAVERSE) {
+    const directoryContents = await fs.readdir(path ?? './');
+
+    const topDir = !path
+      ? [{ path: path ?? './', type: 'dir', name: path ?? './' }]
```

Comment:
I see `path ?? './'` 3 times. Can this be stored as a variable? Perhaps `topPath = path ?? './'`?